### PR TITLE
REST base in admin GUI + DB system parameters management

### PR DESCRIPTION
**Pull request by pjuza**
- added full AOP support (I will add wiki pages about AOP support)
- created base for REST controllers - see _org.openhubframework.openhub.admin.web.configuration.rest.DbConfigurationController_
- created base for unit tests for testing REST endpoints - see _org.openhubframework.openhub.test.rest_ package
- added RPC concept - see _org.openhubframework.openhub.admin.web.common.rpc_ package
- added exception handler _org.openhubframework.openhub.web.common.ExceptionRestHandler_, see https://openhubframework.atlassian.net/browse/OHFJIRA-63 for fault responses
- implemented REST services for configuration parameters, see https://openhubframework.atlassian.net/browse/OHFJIRA-55

Shortly:
- correctly configured AOP support
- created base for REST unit tests
- implemented first REST unit test in admin module - _DbConfigurationControllerTest_
- added implementation of RPC and controller
- added new attribute description to _DbConfigurationParam_
- moved web/config
- introduced _SuperEntity_, parent for all entities
- added exception handler
- added RPC for faults
- corrected profile limitations for H2 server
- added base for REST services, MVC configuration

Issue: OHFJIRA-55, OHFJIRA-63

What is not done?
- i18n support - only English is at this moment
- better data validations in RPC/controllers, for example using _BindingResult_ or JSR-303 Bean Validation API. 
- AOP is configured and entities/services should be refactored to be trully objective

I will continue on these issues in this PR (apart from i18n, it's final for version 2.0).

### CODE REVIEW

**Comment by pjuza** [comment]
I added documentation to wiki:

- getting started - IDE settings: https://openhubframework.atlassian.net/wiki/x/SIM
- AOP - https://openhubframework.atlassian.net/wiki/display/OHF/AOP

File: core-api/src/main/java/org/openhubframework/openhub/api/configuration/DbConfigurationParam.java [hunk 1]
```diff
@@ -68,33 +66,42 @@
...
+    @Column(name = "description", length = 1000)
+    private String description;
+
     @Column(name = "validation", length = 100)
     private String validationRegEx;
 
 
     /** Default public constructor. */
     public DbConfigurationParam() {
+        super(null);
```
Comment (hanusto) [inline core-api/src/main/java/org/openhubframework/openhub/api/configuration/DbConfigurationParam.java:78]: I think that we can provide "default" constructor which reuse parametrized one. This is common scenario.

File: core-api/src/main/java/org/openhubframework/openhub/api/configuration/DbConfigurationParam.java [hunk 2]
```diff
@@ -108,18 +115,41 @@ public DbConfigurationParam(String code, String categoryCode, @Nullable String c
...
 
+    @Override
+    public void setId(@Nullable String s) {
+        throw new UnsupportedOperationException("it's not possible to change identifier of this entity: "
+                + DbConfigurationParam.class.getName());
+    }
+
+    @Nullable
+    @Override
+    public String getId() {
```
Comment (hanusto) [inline core-api/src/main/java/org/openhubframework/openhub/api/configuration/DbConfigurationParam.java:140]: In future, we should use `Optional` pattern.

File: core-api/src/main/java/org/openhubframework/openhub/api/exception/InternalErrorEnum.java [hunk 3]
```diff
@@ -96,11 +96,6 @@
     E112("locking exception - unsuccessful getting lock for the DB record."),
 
     /**
-     * There is no requested invoice in the repository.
```
Comment (hanusto) [inline core-api/src/main/java/org/openhubframework/openhub/api/exception/InternalErrorEnum.java:99]: 👍

File: web-admin/src/main/java/org/openhubframework/openhub/web/common/CorsProperties.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+package org.openhubframework.openhub.web.common;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.cors.CorsConfiguration;
+
+
+/**
+ * WisePorter specific CORS configuration which could be enabled/disabled via {@link #CORS_ENABLED} property.
```
Comment (hanusto) [inline web-admin/src/main/java/org/openhubframework/openhub/web/common/CorsProperties.java]: Wrong javadoc.

File: core/src/test/java/org/openhubframework/openhub/core/DateCompatibilityTest.java [hunk 5]
```diff
@@ -63,16 +64,30 @@ public void testXmlConversion() {
...
+            assertThat(JaxbDateAdapter.printDate(xmlDate), startsWith("2013-10-05+01:00"));
+        }
 
         assertWrongXmlDateTime("2013-10-05");
         assertWrongXmlDateTime("2001-10-26T25:32:52+02:00");
+    }
 
-        OffsetDateTime xmlDate = JaxbDateAdapter.parseDate("2013-10-05");
-        assertThat(JaxbDateAdapter.printDate(xmlDate), startsWith("2013-10-05+01:00"));
+    private boolean isSommerTime() {
```
Comment (hanusto) [inline core/src/test/java/org/openhubframework/openhub/core/DateCompatibilityTest.java]: Summer :)

**Comment by hanusto** [comment]
Thx for this PR.

Only few comments:
- probably corrupted image or link at https://openhubframework.atlassian.net/wiki/display/OHF/Getting+started
- https://openhubframework.atlassian.net/wiki/display/OHF/AOP: I dont understand why wiseporter code is used and referenced? It is not clear for foreign. We have to rewrite that.
- I think that we can use CORS config from Spring Boot
- `SuperEntity` can provided default constructor.

Now I approve that with merge.

**Review by hanusto — APPROVED** [review]

**Pull request merged**